src/lib/server/pages.rs

```diff
@@ -541,3 +545,68 @@ pub async fn unauthenticate_onvif_device(
 
     Ok(HttpResponse::Ok().finish())
 }
+
+#[api_v2_operation]
+/// WebSocket endpoint that streams the DOT representation of all running GStreamer pipelines in real time.
+pub async fn dot_stream(req: HttpRequest, stream: web::Payload) -> Result<HttpResponse> {
+    let (response, mut session, mut msg_stream) =
+        actix_ws::handle(&req, stream).map_err(|error| Error::Internal(format!("{error:?}")))?;
+
+    rt::spawn(async move {
+        let mut interval = tokio::time::interval(Duration::from_secs(2));
+        let mut last_dots = std::collections::HashMap::new();
+
+        loop {
+            tokio::select! {
+                Some(msg) = msg_stream.next() => {
+                    match msg {
+                        Ok(Message::Ping(_)) | Ok(Message::Pong(_)) | Ok(Message::Text(_)) | Ok(Message::Binary(_)) | Ok(Message::Continuation(_)) | Ok(Message::Nop) => continue,
+                        Ok(Message::Close(_)) | Err(_) => break,
+                    }
+                }
+                _ = interval.tick() => {
+                    let mut dots = Vec::new();
+                    let mut changed = false;
+
+                    match crate::stream::manager::streams().await {
+                        Ok(streams) => {
+                            for stream_info in streams {
+                                if let Some((dot, childrens)) = crate::stream::manager::Manager::get_stream_dot_by_id(&stream_info.id).await {
+                                    let id = stream_info.id.to_string();
+                                    if last_dots.get(&id).map_or(true, |last_dot| last_dot != &dot) {
+                                        last_dots.insert(id.clone(), dot.clone());
+                                        changed = true;
+                                        dots.push(json!({
+                                            "id": id,
+                                            "dot": dot,
+                                            "childrens": childrens,
```

Comment:
"children" is already plural :sweat_smile: 
```suggestion
                                            "children": children,
```